kratos/solving_strategies/schemes/residual_based_implicit_time_scheme.h

```diff
@@ -158,43 +166,41 @@ class ResidualBasedImplicitTimeScheme
 
     /**
      * @brief It initializes a non-linear iteration (for an individual condition)
-     * @param pCondition The condition to compute
+     * @param pCurrentCondition The condition to compute
      * @param rCurrentProcessInfo The current process info instance
      */
-
     void InitializeNonLinearIteration(
-        Condition::Pointer pCondition,
+        Condition::Pointer pCurrentCondition,
         ProcessInfo& rCurrentProcessInfo
         ) override
     {
-        pCondition->InitializeNonLinearIteration(rCurrentProcessInfo);
+        pCurrentCondition->InitializeNonLinearIteration(rCurrentProcessInfo);
     }
     
     /**
      * @brief It initializes a non-linear iteration (for an individual element)
-     * @param pElement The element to compute
+     * @param pCurrentElement The element to compute
      * @param rCurrentProcessInfo The current process info instance
      */
-
     void InitializeNonLinearIteration(
-        Element::Pointer pElement,
+        Element::Pointer pCurrentElement,
```

Comment:
probably out of this pr scope ... is there a reason why this functions is not templated ?

Reply:
a template function cannot be virtual